plugins/processors/clone/clone.go

```diff
@@ -23,25 +23,32 @@ func (*Clone) SampleConfig() string {
 }
 
 func (c *Clone) Apply(in ...telegraf.Metric) []telegraf.Metric {
-	cloned := []telegraf.Metric{}
+	out := make([]telegraf.Metric, 0, 2*len(in))
 
-	for _, metric := range in {
-		cloned = append(cloned, metric.Copy())
+	for _, raw := range in {
+		// Strip the tracking information if the metric is wrapped.
+		original := raw
+		if wm, ok := raw.(telegraf.UnwrappableMetric); ok {
+			original = wm.Unwrap()
+		}
```

Comment:
Like starlark, if we deepcopy a metric that creates a second metric that we are expecting to exist at the output, so we should maintain the tracking information end-to-end for the new cloned metric as well.